great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -52,69 +54,63 @@ class SQLDatasourceWarning(UserWarning):
     pass
 
 
-@pydantic_dc.dataclass(frozen=True)
-class ColumnSplitter:
+class _ColumnSplitter(ExperimentalBaseModel):
     column_name: str
     method_name: str
 
+    @property
+    def param_names(self) -> List[str]:
+        raise NotImplementedError
+
+    def splitter_method_kwargs(self) -> Dict[str, Any]:
+        raise NotImplementedError
+
+    def request_options_to_batch_spec_kwarg_identifiers(
+        self, options: BatchRequestOptions
+    ) -> Dict[str, Any]:
+        """Translate the batch request options to the dictionary needed to by our execution engine
+
+        Arguments:
+            options: A BatchRequest.options dictionary that specified all the fields necessary
+                     to specify a batch with respect to this column splitter.
+
+        Returns:
+            A dictionary that can be added to batch_spec_kwargs["batch_identifiers"] when we
+            make a SqlAlchemyDatasourceBatchSpec and run execution_engine.get_batch_data_and_markers
+            for a particular batch specified by options.
+        """
+        raise NotImplementedError
+
     def param_defaults(self, sql_asset: _SQLAsset) -> List[Dict]:
         from great_expectations.execution_engine.split_and_sample.sqlalchemy_data_splitter import (
             SqlAlchemyDataSplitter,
         )
 
         execution_engine = sql_asset.datasource.get_execution_engine()
         splitter = SqlAlchemyDataSplitter(execution_engine.dialect_name)
-        batch_identifiers = splitter.get_data_for_batch_identifiers(
+        batch_identifier_data = splitter.get_data_for_batch_identifiers(
             execution_engine=execution_engine,
             selectable=sql_asset.as_selectable(),
             splitter_method_name=self.method_name,
-            splitter_kwargs={"column_name": self.column_name},
+            splitter_kwargs=self.splitter_method_kwargs(),
         )
         params: List[Dict] = []
-        for batch_identifier in batch_identifiers:
-            # batch_identifiers is a list of dicts where each dict contains
-            # a column name key whose value is the batch parameters
-            params.append(batch_identifier[self.column_name])
+        for identifier_data in batch_identifier_data:
+            # batch_identifiers is a list of dicts that is keyed by the column name. The value is either:
+            #  * A dict whose keys are batch request parameter names and values are the batch values.
+            #  * A single value for the 1 batch request parameter.
+            value = identifier_data[self.column_name]
+            if isinstance(value, dict):
+                params.append(value)
+            elif len(self.param_names) == 1:
+                params.append({self.param_names[0]: value})
+            else:
+                raise SQLDatasourceError(
+                    f"identifier_data is in an unexpected form for {self.__class__} "
+                    f"on column {self.column_name}: {identifier_data}"
```

Comment:
@billdirks Same question as just above -- where would the user go to learn the expected form of the "identifier_data"?  Thanks.

Reply:
No user should ever need to worry about the form of `identifier_data`. This is an internal implementation detail and is a shim to our current system splitter code which is located [here](https://github.com/great-expectations/great_expectations/blob/f0e89bdbaa491f39ba06f3d7af3d0b4d31d425bf/great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py)